vault/identity_store_oidc.go

```diff
@@ -1791,6 +1794,7 @@ func (i *IdentityStore) expireOIDCPublicKeys(ctx context.Context, s logical.Stor
 			if err := s.Delete(ctx, publicKeysConfigPath+keyID); err != nil {
 				i.Logger().Error("error deleting OIDC public key", "key_id", keyID, "error", err)
 				nextExpiration = now
+				continue
```

Comment:
This makes me think whether we should move L1793 (`didUpdate = true`) to be called on L1799 instead. This way we only flush the cache on successful deletion of the key(s).

Reply:
I was just thinking to do this too 👍  Done in 8da0631.